File: Bridge/Vasylyshyn/src/Shape/Circle.java
package Shape;

import Color.Color;

public class Circle extends Shape {
    public int r;

    public Circle(Color color, int r) {
        super( color);
        this.r=r;
    }

    public Circle() {
    }

    @Override
    public String toString() {
        return "Circle{" +
                "r=" + r +
                ", color=" + color +
                '}';
    }

    @Override
    public double Area() {
        return this.r*Math.PI*Math.PI;
    }

    @Override
    public double Perimeter() {
        return 2*Math.PI*this.r;
    }

    @Override
    public void draw() {
        System.out.println("It is Circle. It is already painted");
        color.Paint();

    }
}

File: Bridge/KovalchukD-03/ConsoleApp25/circle.cs
using System;

namespace BridgePattern
{
    class Circle : Shape
    {
        private double radius;

        public Circle(double radius, string color) : base(color)
        {
            if (radius <= 0)
            {
                throw new ArgumentException("Radius must be a positive number.");
            }

            this.radius = radius;
        }

        public override double Area()
        {
            return Math.Round(Math.PI * radius * radius, 3);
        }

        public override double Perimeter()
        {
            return Math.Round(2 * Math.PI * radius, 3);
        }

        public override void Draw()
        {
            Console.WriteLine($"Circle: Perimeter={Perimeter()}, Area={Area()}, Color={color}");
        }
    }
}
